scrapy/extensions/feedexport.py

```diff
@@ -397,6 +407,15 @@ def _close_slot(self, slot, spider):
         d.addErrback(
             self._handle_store_error, logmsg, spider, type(slot.storage).__name__
         )
+
+        self.pending_deferreds.append(d)
+        d.addCallback(
+            lambda _: self.crawler.signals.send_catch_log(
+                signals.feed_slot_closed, slot=slot
+            )
```

Comment:
No strong opinion on it, but I wonder if it would not make sense to make this signal also awaitable, like `feed_exporter_closed`, since it might not be that much work.

Reply:
I addressed the changes above and this one. I can't think of any other change than using send_catch_log_deferred, I added some tests and it works. Is that what you were referring to?